libs/gretty/src/main/groovy/org/akhikhl/gretty/ProjectUtils.groovy

```diff
@@ -83,22 +84,45 @@ final class ProjectUtils {
         addProjectClassPathWithOverlays(urls, proj.project(overlay), dependencyConfigName)
   }
 
-  private static void addProjectClassPathWithDependencies(Set<URL> urls, Project proj, String dependencyConfigName) {
-    if(proj.sourceSets.main.output.files.any { new File(it, 'META-INF/web-fragment.xml').exists() })
-      urls.add proj.jar.archivePath.toURI().toURL()
-    else
-      urls.addAll proj.sourceSets.main.output.files.collect { it.toURI().toURL() }
-    def dependencyConfig = proj.configurations.findByName(dependencyConfigName)
-    if(dependencyConfig) {
-      def files = dependencyConfig.fileCollection { !(it instanceof ProjectDependency) }
-      def grettyProvidedCompileConfig = proj.configurations.findByName('grettyProvidedCompile')
-      if(grettyProvidedCompileConfig)
-        files = files - grettyProvidedCompileConfig
-      urls.addAll(files.collect { it.toURI().toURL() })
-      dependencyConfig.allDependencies.withType(ProjectDependency).each { dep ->
-        addProjectClassPathWithDependencies(urls, dep.getDependencyProject(), dependencyConfigName)
+  private static void addProjectClassPathWithDependencies(Set<URL> urls, Project project, String dependencyConfigName) {
+    Set<File> files = []
+
+    def (outputs, archives) = collectOutputsAndArchivesWithDependencies(project, dependencyConfigName)
+    files += outputs
+    files += project.configurations.findByName(dependencyConfigName).collect().toSet()
+    files -= archives
```

Comment:
@f4lco Thanks for your review. I'm not so familiar with Groovy and I think your recommendations are very solid.

I prefer to put the project output folders in front of the external dependencies. So, I did a slight change based on your recommendations. Others are near as the same as your recommendations.

Reply:
That's fine with me. Thank you for being so open to suggestions 👍